wheatley/bot.py

```diff
@@ -81,8 +82,10 @@ def __init__(self, tower: RingingRoomTower, row_generator: RowGenerator, do_up_d
 
         self._row_number = 0
         self._place = 0
-        self._rounds: Row = rounds(MAX_BELL)
+        self._opening_rounds: Row = row_generator.start_row
+        self._rounds: Row = rounds(self.number_of_bells)
```

Comment:
Is `self.number_of_bells` well defined at this point? I have a feeling `_on_size_change` is called once the tower is connected and joined the tower, so this is more a place holder

Reply:
your right, I call generate_starting_row in _on_size_change for that reason. 
I'm happy to switch back to rounds(MAX_BELL) not sure it matters too much?